lib/matplotlib/image.py

```diff
@@ -390,22 +390,24 @@ def _make_image(self, A, in_bbox, out_bbox, clip_bbox, magnification=1.0,
                 # float64's ability to represent changes.  Applying
                 # a norm first would be good, but ruins the interpolation
                 # of over numbers.
-                if self.norm.vmin is not None and self.norm.vmax is not None:
-                    dv = (np.float64(self.norm.vmax) -
-                          np.float64(self.norm.vmin))
-                    vmid = self.norm.vmin + dv / 2
-                    newmin = vmid - dv * 1.e7
-                    if newmin < a_min:
-                        newmin = None
-                    else:
-                        a_min = np.float64(newmin)
-                    newmax = vmid + dv * 1.e7
-                    if newmax > a_max:
-                        newmax = None
-                    else:
-                        a_max = np.float64(newmax)
-                    if newmax is not None or newmin is not None:
-                        A_scaled = np.clip(A_scaled, newmin, newmax)
+                vmin = self.norm.vmin if self.norm.vmin is not None else a_min
```

Comment:
I would do `if self.norm.vmin is None or self.norm.vmax is None: self.norm.autoscale_None(A)`

Reply:
Sure - its OK to modify the norm here?

Reply:
don't even need the if-statement in that case.